### ISSUE TITLE
Always meet a bug when apply to my dataset

### ISSUE THREAD

**Issue by ElegantLin** [gdewael/cpg-transformer#4]
Hi, @gdewael ,

Thanks for your great work. I am using the pre-trained model to impute my dataset. However, when I am executing `dm.setup(None)`. I always met a bug that the out-of-range exception in the `cuts = torch.tensor([(indices[i],indices[i+self.segment_size-1]) for i in cuts_])` given [here](https://github.com/gdewael/cpg-transformer/blob/main/cpg_transformer/datamodules.py#L241) and the code would go to the exception branch because my indices cannot cover the whole the chromosome due to the mask. Do you have some suggestions?

Thanks!

**Comment by gdewael** [comment]
Hi! Thanks for trying our code.

I implemented this exception to catch the case in which a chromosome has fewer CpG sites ([computed here](https://github.com/gdewael/cpg-transformer/blob/970248d192739af194c9e9df5823c9eda8c2fa75/cpg_transformer/datamodules.py#L238)) than expected to make a whole segment (a segment is what is input to the model all at once, kind of like how you would cut up sentences to input in NLP). If this exception is triggered, it will just make one segment for the whole chromosome. So in essence, this exception is not an error.

If this doesn't solve your issue, is there a way to generate some toy data to reproduce the issue?

Best, Gaetan

**Comment by ElegantLin** [comment]
Hi, @gdewael 

Thanks for your quick reply. I think I have found the issue, which is from my code rather than your code. I would close this issue after I verify all my chromosomes.

Thanks a lot!
